### DISCUSSION TITLE
In Latest 0.1.4, do I need mounted anymore around my Menus?

### DISCUSSION THREAD

**Discussion by httpete** [ariakit/ariakit#2315] [Questions] [answered]
This was in all the original examples, but I don't see it in the latest ones.

`{menu.mounted && (`

**Comment by diegohaz** [comment, marked as answer]
You can dynamically render the popover based on the `mounted` state, just as before. Nothing has changed in that regard except in how you access the `mounted` state:

```diff
- const mounted = state.mounted;
+ const mounted = store.useState("mounted");
```

Some examples that dynamically render the popover:

- [Sliding menu](https://ariakit.org/examples/menu-slide)
- [MenuBar](https://ariakit.org/examples/menu-bar)
- [Standalone Popover](https://ariakit.org/examples/popover-standalone) (doesn't use the mounted state, but dynamically render the popover)
- [Comboobx with links](https://ariakit.org/examples/combobox-links)
- [Textarea with inline Combobox](https://ariakit.org/examples/combobox-textarea)
- [Multi-Select](https://ariakit.org/examples/select-multiple)

**Reply by httpete** [reply]
Thank you @diegohaz . Do we need to do the menu.mounted though for performance? It seems fine without it but we have very deep menus.

**Reply by diegohaz** [reply]
You usually don't need it. But if you encounter any performance issues, dynamically rendering the contents may help with the initial render/hydration speed. But in this case, I think I'd just wrap the contents in a Suspense boundary.

**Reply by httpete** [reply]
In my case I have a multi page app, which loads the menus on each load. So speed to render here is crucial. Thank you.

**Reply by httpete** [reply]
What is the distinction between getState and useState?

**Reply by diegohaz** [reply]
https://ariakit.org/guide/component-stores#reading-the-state-on-events

Let me know if we should improve it.

**Reply by httpete** [reply]
@diegohaz , yes I think it would be an improvement in the doc here:

_In addition to state, component stores may also accept state updater functions. These functions are called with the new state whenever the state is updated. You can use them for various purposes, such as updating another state or performing side effects._

They really are not state updater functions, the functions themselves don't update the state. They are callback events when state HAS been updated. This confused me in both of my discussions and I think it would confuse others as well.

I would call them "state change callbacks".

**Reply by diegohaz** [reply]
@httpete That makes sense. Would you like to submit a PR?

**Reply by httpete** [reply]
Diego, I cloned and I am running the localhost:3000, but I can't see how to make edits. I will fix the instructions here too once you show me.
https://github.com/ariakit/ariakit/tree/main/website

**Reply by diegohaz** [reply]
You can edit the markdown files here: https://github.com/ariakit/ariakit/tree/main/guide

You can also check the contributing guide for more details: https://github.com/ariakit/ariakit/blob/main/contributing.md
